boot/bootutil/src/bootutil_misc.c

```diff
@@ -69,52 +68,69 @@ struct boot_swap_table {
  */
 static const struct boot_swap_table boot_swap_tables[] = {
     {
-        .magic_slot0 =      0,
+        .magic_slot0 =      BOOT_MAGIC_ANY,
         .magic_slot1 =      BOOT_MAGIC_GOOD,
-        .image_ok_slot0 =   0,
-        .image_ok_slot1 =   0xff,
-        .copy_done_slot0 =  0,
+        .image_ok_slot0 =   BOOT_FLAG_ANY,
+        .image_ok_slot1 =   BOOT_FLAG_UNSET,
+        .copy_done_slot0 =  BOOT_FLAG_ANY,
         .swap_type =        BOOT_SWAP_TYPE_TEST,
     },
     {
-        .magic_slot0 =      0,
+        .magic_slot0 =      BOOT_MAGIC_ANY,
         .magic_slot1 =      BOOT_MAGIC_GOOD,
-        .image_ok_slot0 =   0,
-        .image_ok_slot1 =   0x01,
-        .copy_done_slot0 =  0,
+        .image_ok_slot0 =   BOOT_FLAG_ANY,
+        .image_ok_slot1 =   BOOT_FLAG_SET,
+        .copy_done_slot0 =  BOOT_FLAG_ANY,
         .swap_type =        BOOT_SWAP_TYPE_PERM,
     },
     {
         .magic_slot0 =      BOOT_MAGIC_GOOD,
         .magic_slot1 =      BOOT_MAGIC_UNSET,
-        .image_ok_slot0 =   0xff,
-        .image_ok_slot1 =   0,
-        .copy_done_slot0 =  0x01,
+        .image_ok_slot0 =   BOOT_FLAG_UNSET,
+        .image_ok_slot1 =   BOOT_FLAG_ANY,
+        .copy_done_slot0 =  BOOT_FLAG_SET,
         .swap_type =        BOOT_SWAP_TYPE_REVERT,
     },
 };
 
 #define BOOT_SWAP_TABLES_COUNT \
     (sizeof boot_swap_tables / sizeof boot_swap_tables[0])
 
-int
-boot_magic_code(const uint32_t *magic)
+static int
+boot_magic_code(const struct flash_area *fap, const uint32_t *magic)
```

Comment:
I think the `_code` naming is very appropriate here but I understand it might be misinterpreted so maybe `boot_magic_decode` and `boot_flag_decode` would be better.

Reply:
yes!